src/raftstore/store/peer.rs

```diff
@@ -1438,87 +1342,75 @@ impl Peer {
         self.proposals.gc();
     }
 
+    fn response_read<T, C>(
+        &mut self,
+        mut read: ReadIndexRequest,
+        ctx: &mut PollContext<T, C>,
+        replica_read: bool,
+    ) {
+        debug!(
+            "handle reads with a read index";
+            "request_id" => ?read.binary_id(),
+            "region_id" => self.region_id,
+            "peer_id" => self.peer.get_id(),
+        );
+        RAFT_READ_INDEX_PENDING_COUNT.sub(read.cmds.len() as i64);
+        for (req, cb) in read.cmds.drain(..) {
+            if !replica_read {
+                cb.invoke_read(self.handle_read(ctx, req, true, read.read_index));
+                continue;
+            }
+            if req.get_header().get_replica_read() {
```

Comment:
Does `!replica_read || req.get_header().get_replica_read()` always be true? If it is, I think we could remove all the conditions.

Reply:
I'm not sure. `replica_read` indicates the peer currently is follower. Maybe there could be some requests proposed when the peer is leader, and the peer steps down to follower and then get here?

Reply:
For this purpose, I think checking the leadership there may be a better way.

Reply:
Better add a comment.